### DISCUSSION TITLE
valkey default configs

### DISCUSSION THREAD

**Discussion by amirreza-sf80** [valkey-io/valkey#1280] [Q&A]
hi
i want to write a config file
in my time exploring i found [this comment](https://github.com/valkey-io/valkey-container/blob/mainline/8.0/debian/Dockerfile#L55)

from what i understand if i write a conf file i have to specify all the configs

now i'm just writing the conf for tests (need unix sockets) and don't need anything fancy

so thought i could cope paste the default settings

but i can't find the default settings anywhere, well they are in the source code (i persume) but going through C code is not easy for me

if anyone can provide me a list of default settings i'd be thankful.

**Comment by madolson** [comment]
No, if you specify a config file it will take the default values except those that you override in the config file you specify. If you have a config file that just has `port 6500`, it will only override that specific configuration.

@roshkhatri Can you create an issue to remove the [hijinks in our build script that was linked to](https://github.com/valkey-io/valkey-container/blob/mainline/8.0/debian/Dockerfile#L55)? It seems very odd we do that and wrong?

**Comment by amirreza-sf80** [comment]
oh, ok thanks

but is there any list to see all the defalt values?
i know about the valkey.conf doc, but that's very long and hard to read

**Reply by zuiderkwast** [reply]
We would like a proper documentation page with all the configs but nobody has done it yet.
